### PR TITLE
api: fix shared repos filtering and permissions.

### PR DESCRIPTION
**Pull request by alejandromumo**
closes https://github.com/zenodo/rdm-project/issues/427

### CODE REVIEW

File: invenio_github/views/github.py [hunk 1]
```diff
@@ -146,8 +151,11 @@ def get_repository(repo_name):
             )
         except RepositoryAccessError:
             abort(403)
-        except NoResultFound:
+        except (NoResultFound, RepositoryNotFoundError):
             abort(404)
+        except Exception as exc:
+            current_app.logger.exception(str(exc))
+            abort(400)
```
Comment (alejandromumo) [inline invenio_github/views/github.py:158]: We discussed about adding error handlers that could use our default error handling. That way, we could benefit from sending the sentry error ID in the response and that could help us debug issues faster.

However, that means we need to change a bit the code here. 

In order not to block the fix for shared repos, I will tackle that [separately](https://github.com/inveniosoftware/invenio-github/issues/156).

File: invenio_github/api.py [hunk 2]
```diff
@@ -347,7 +361,12 @@ def get_user_repositories(self):
         repos = deepcopy(self.user_available_repositories)
         if repos:
             # 'Enhance' our repos dict, from our database model
-            for repo in self.user_enabled_repositories:
+            db_repos = Repository.query.filter(
+                Repository.github_id.in_(
+                    [int(k) for k in self.user_available_repositories.keys()]
```
Comment (alejandromumo) [inline invenio_github/api.py:366]: We were filtering repos based on `self.user_enabled_repositories`, which means "repositories enabled locally by the user". That was wrong

File: invenio_github/api.py [hunk 3]
```diff
@@ -243,13 +259,14 @@ def sync_repo_hook(self, repo_id):
         # If hook on GitHub exists, get or create corresponding db object and
         # enable the hook. Otherwise remove the old hook information.
         repo = Repository.get(repo_id, gh_repo.full_name)
-        if not repo:
-            repo = Repository.create(self.user_id, repo_id, gh_repo.full_name)
 
         if hook:
+            if not repo:
```
Comment (alejandromumo) [inline invenio_github/api.py:264]: This means that we were creating one local repo per repository that users had in `remote_account.extra_data`. Therefore, we have too many repos now created locally.

There is no big impact but we can clean this up. I created an [issue](https://github.com/zenodo/rdm-project/issues/454) for that

File: invenio_github/api.py [hunk 4]
```diff
@@ -107,6 +99,30 @@ def session_token(self):
     )
     """Return OAuth remote application."""
 
+    def check_repo_access_permissions(self, repo):
+        """Checks permissions from user on repo.
```
Comment (alejandromumo) [inline invenio_github/api.py:103]: After discussing with @slint , we decided to change how the permissions on a repo are computed. 

Namely, an user can read / enable / disable / hook a repo if the user is the owner OR has admin access in GitHub.

File: invenio_github/api.py (hunk 3 above)
Comment (kpsherva) [reply invenio_github/api.py:264]: could you add a comment in the code?

**Pull request merged**